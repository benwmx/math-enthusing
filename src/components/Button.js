/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import React from 'react';

const Button = props => {
  const generateClassName = name => {
    let className = 'button ';
    if (Number.isNaN(parseInt(name, 10)) && (name !== '.' && name !== 'AC' && name !== '%' && name !== '+/-')) className += 'operator';
    if (name === '0') className += 'zero';
    return className;
  };
  return (
    <button
      type="button"
      name={props.name}
      className={generateClassName(props.name)}
      onClick={props.runCalculation}
    >
      {props.name}
    </button>
  );
};

export default Button;
