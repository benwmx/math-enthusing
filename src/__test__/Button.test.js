import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

describe('Calculator Component test', () => {
  it('Should render ', () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
