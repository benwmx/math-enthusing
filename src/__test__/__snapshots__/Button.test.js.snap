// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component test Should render  1`] = `
<button
  className="button operator"
  type="button"
/>
`;
